Sources/CvMap.cpp

```diff
@@ -966,35 +879,19 @@ int CvMap::getMapFractalFlags()
 
 
 //"Check plots for wetlands or seaWater.  Returns true if found"
-bool CvMap::findWater(CvPlot* pPlot, int iRange, bool bFreshWater)
+bool CvMap::findWater(CvPlot* pPlot, int iRange, bool bFreshWater) const
 {
 	PROFILE("CvMap::findWater()");
 
-	CvPlot* pLoopPlot;
-	int iDX, iDY;
-
-	for (iDX = -(iRange); iDX <= iRange; iDX++)
+	for (int iDX = -(iRange); iDX <= iRange; iDX++)
```

Comment:
We need to make a `plot_range_iterator` that takes x,y and range to clean up all these loops. 
Then we could just do `std::find(plot_range(x, y, range), plot_range_end(), _1->isFreshWater())`.
Not a comment for you to change anything just a notification that I am going to add one and it will mean these places all get refactored to use it anyway, so don't worry about them too much. This goes for anywhere that is doing ranged algorithm type loops that could be replaced by an iterator type and an stl algorithm call.